.jenkins/cscs-perftests/env-perftests.sh

```diff
@@ -4,14 +4,10 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-cxx_std="20"
```

Comment:
Because that's what we were using before as well (accidentally), see e.g. https://lisone.cscs.ch/job/ssl/job/pika%20performance%20tests/2974/artifact/pika/jenkins-pika-perftests.err/*view*/. This is the part I was referring to with the environment being changed a bit (i.e. the CMake options set in `env-perftests.sh` were never taken into account so I've removed them to avoid confusion; the spec is still used). Since we only use header-only boost I'm guessing the `cxxstd` variant on the package didn't make a difference.